workflow/JSON-visualizer/index.js

```diff
@@ -0,0 +1,93 @@
+const jsonWrapper = document.querySelector('.json-wrapper');
+const buildTreeButton = document.querySelector('.build-tree-btn');
+
+function isJson(str) {
+  try {
+    JSON.parse(str);
+  } catch (e) {
+    return false;
+  }
+  return true;
+}
+
+function createAndAppendParagraph(parent, innerHtml) {
+  const newParagraph = document.createElement('p');
+  newParagraph.innerHTML = innerHtml;
+  parent.append(newParagraph);
+}
+
+function createAndAppendDivWithClass(parent, className) {
+  const div = document.createElement('div');
+  div.className = className;
+  parent.append(div);
+  return div;
+}
+
+function highlightDifferentDataTypes(data) {
+  switch (typeof data) {
+    case 'number':
+      return 'red';
+
+    case 'string':
+      return 'purple';
+
+    case 'bigint':
+      return 'yellow';
+
+    case 'boolean':
+      return 'green';
+
+    default:
+      return 'blue';
+  }
+}
+
+function addTypeWrapper(valueToCheckType, valueToReturn = valueToCheckType) {
+  switch (typeof valueToCheckType) {
+    case 'string':
+      return `"${valueToReturn}"`;
+
+    case 'object':
+      if (!valueToReturn) return valueToReturn;
+
+      return Array.isArray(valueToCheckType) ? `[${valueToReturn}]` : `{${valueToReturn}}`;
+
+    default:
+      return valueToReturn;
+  }
+}
+
+function buildJsonTree(jsonStr, parentDiv) {
+  for (let key in jsonStr) {
+    if (!!jsonStr[key] && typeof jsonStr[key] === 'object') {
```

Comment:
you don't need to write `!!` in if expression, all truthy expressions will be taken as `true` and falsy - `false`